source/includes/write/replace.cpp

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::basic::kvp;
+using bsoncxx::builder::basic::make_document;
+
+int main() {
+    mongocxx::instance instance;
+    mongocxx::uri uri("<connection string>");
+    mongocxx::client client(uri);
+
+    // start-db-coll
+    auto db = client["sample_restaurants"];
+    auto collection = db["restaurants"];
+    // end-db-coll
+
+    {
+        // replaces a document that has a "name" value of "Nobu"
+        // with a document that has a "name" of "La Bernadin"
+        // start-replace-one
+        auto query_filter = make_document(kvp("name", "Nobu"));
+        auto replace_doc = make_document(kvp("name", "La Bernadin"));
+
+        auto result = collection.replace_one(query_filter.view(), replace_doc.view());
+        // end-replace-one
+    }
+
+    {
+        // Retrieves a document to verify the replacement operation
+        // start-replace-one-io
+        auto new_doc = collection.find_one(make_document(kvp("name", "La Bernadin")));
+        std::cout << "New document: " << bsoncxx::to_json(*new_doc) << std::endl;
+        // end-replace-one-io
+    }
+
+    {
+        // start-replace-options-hint
+        auto index_specification = make_document(kvp("name", 1));
+        collection.create_index(index_specification.view());
+
+        mongocxx::options::replace opts{}; 
+        opts.hint(mongocxx::hint{"name_1"});
+
+        auto query_filter = make_document(kvp("name", "Nobu"));
+        auto replace_doc = make_document(kvp("name", "La Bernadin"));
+
+        auto result = collection.replace_one(query_filter.view(), replace_doc.view(), opts);
+        // end-replace-options-hint 
+    }
+    
+    {
+        // Replaces a document that has a "name" value of "Nobu" and instructs the operation to use the "name_1" field index
```

Comment:
S: I'd change this either to the "name" field index or the "name_1" index (since it's an index on the name field, but the index itself is called name_1)
```suggestion
        // Replaces a document that has a "name" value of "Nobu" and instructs the operation to use the "name" field index
```